code/game/machinery/newscaster/newscaster_machine.dm

```diff
@@ -0,0 +1,551 @@
+				return
+			var/questionable_message = params["messageID"]
+			for(var/datum/newscaster/feed_message/mess in current_channel.messages)
+				if(mess.message_ID == questionable_message)
+					mess.toggleCensorBody()
+					break
+
+		if("authorCensor")
+			if (!params["secure"])
+				say("Clearance not found.")
+				return
+			var/questionable_message = params["messageID"]
+			for(var/datum/newscaster/feed_message/mess in current_channel.messages)
+				if(mess.message_ID == questionable_message)
+					mess.toggleCensorAuthor()
+					break
+
+		if("channelDNotice")
+			if (!params["secure"])
+				say("Clearance not found.")
+				return
+			var/proto_chan = (params["channel"])
+			for(var/datum/newscaster/feed_channel/potential_channel in GLOB.news_network.network_channels)
+				if(proto_chan == potential_channel.channel_ID)
+					current_channel = potential_channel
+					break
+			current_channel.toggleCensorDclass()
+
+		if("printNewspaper")
+			if(paper_remaining <= 0)
+				balloon_alert_to_viewers("Unit out of Paper!")
+				return
+			print_paper()
+
+		//**************************
+		//	  Bounty Board Acts
+		//**************************
+		if("createBounty")
+			if(!current_user || !bounty_text)
+				playsound(src, 'sound/machines/buzz-sigh.ogg', 20, TRUE)
+				return TRUE
+			for(var/datum/station_request/i in GLOB.request_list)
+				if("[i.req_number]" == "[current_user.account_id]")
+					say("Account already has active bounty.")
+					return
+			var/datum/station_request/curr_request = new /datum/station_request(current_user.account_holder, bounty_value,bounty_text,current_user.account_id, current_user)
+			GLOB.request_list += list(curr_request)
+			for(var/obj/i in GLOB.allbountyboards)
+				balloon_alert_to_viewers("New bounty has been added!")
+				playsound(i.loc, 'sound/effects/cashregister.ogg', 30, TRUE)
+		if("apply")
+			if(!current_user)
+				say("Please equip a valid ID first.")
+				return TRUE
+			if(current_user.account_holder == active_request.owner)
+				playsound(src, 'sound/machines/buzz-sigh.ogg', 20, TRUE)
+				return TRUE
+			for(var/new_apply in active_request?.applicants)
+				if(current_user.account_holder == active_request?.applicants[new_apply])
+					playsound(src, 'sound/machines/buzz-sigh.ogg', 20, TRUE)
+					return TRUE
+			active_request.applicants += list(current_user)
+		if("payApplicant")
+			if(!current_user)
+				return
+			if(!current_user.has_money(active_request.value) || (current_user.account_holder != active_request.owner))
+				playsound(src, 'sound/machines/buzz-sigh.ogg', 30, TRUE)
+				return
+			request_target.transfer_money(current_user, active_request.value)
+			say("Paid out [active_request.value] credits.")
+			return TRUE
+		if("clear")
+			if(current_user)
+				current_user = null
+				say("Account Reset.")
+				return TRUE
+		if("deleteRequest")
+			if(!active_request || !current_user)
+				playsound(src, 'sound/machines/buzz-sigh.ogg', 20, TRUE)
+				return FALSE
```

Comment:
You should basically always return TRUE unless you have a good reason, because it updates the UI.